include/l3.h

```diff
@@ -55,14 +55,57 @@ int l3_init(const char *path);
 
 /**
  * \brief Caller-macro to invoke L3 simple logging.
+ *
+ * Macro to synthesize LOC-encoding value, if needed, under conditional
+ * compilation flags. DEBUG-version of macros are provided to cross-check
+ * that the supplied format string in `msg` are consistent with the
+ * arguments' type.
  */
-#ifdef L3_LOC_ENABLED
-#define l3_log_simple(msg, arg1, arg2)                          \
-        l3__log_simple(__LOC__, (msg), (arg1), (arg2))
-#else   // L3_LOC_ENABLED
-#define l3_log_simple(msg, arg1, arg2)                          \
-        l3__log_simple(L3_ARG_UNUSED, (msg), (arg1), (arg2))
-#endif  //
+#if defined(DEBUG)
+
+  #ifdef L3_LOC_ENABLED
+
+    #define l3_log_simple(msg, arg1, arg2)                              \
+            if (1) {                                                    \
+                l3__log_simple((msg),                                   \
+                               (uint64_t) (arg1), (uint64_t) (arg2),    \
+                                 __LOC__);                              \
+            } else if (0) {                                             \
+                printf((msg), (arg1), (arg2));                          \
+            } else
```

Comment:
To ensure that a call to the macro gets rightly terminated as : `if () ... else;`